src/main/scala/org/phenoscape/kb/Gene.scala

```diff
@@ -61,20 +61,33 @@ object Gene {
       result <- App.executeSPARQLQuery(query)
     } yield ResultCount(result)
 
-  def buildSearchQuery(text: String): Query = {
+  def buildSearchQuery(text: String, limit: Option[Int]): Query = {
     val searchText = if (text.endsWith("*")) text else s"$text*"
-    val query = select_distinct('gene, 'gene_label, 'taxon, 'taxon_label) from "http://kb.phenoscape.org/" where bgp(
-      t('gene_label, BDSearch, NodeFactory.createLiteral(searchText)),
-      t('gene_label, BDMatchAllTerms, NodeFactory.createLiteral("true")),
-      t('gene_label, BDRank, 'rank),
-      t('gene, rdfsLabel, 'gene_label),
-      t('gene, rdfType, AnnotatedGene),
-      t('gene, in_taxon, 'taxon),
-      t('taxon, rdfsLabel, 'taxon_label)
-    )
-    query.addOrderBy('rank, Query.ORDER_ASCENDING)
-    query.setLimit(100)
-    query
+    val queryLimit = limit.map(l => sparql"LIMIT $l").getOrElse(sparql"")
+    val query =
+      sparql"""
+            SELECT DISTINCT ?gene ?gene_label ?taxon ?taxon_label
+            FROM $KBMainGraph
+            WHERE {
+              ?gene_label $BDSearch $searchText .
+              ?gene_label $BDMatchAllTerms true .
```

Comment:
Does the literal `true` need to be `Boolean` or `String`?

Reply:
It is automatically a boolean if written this way.